### PR TITLE
Tanvir dev -> Development

### PR DESCRIPTION
**Pull request by Tanvir-Rahman123**
Pull request to development. 
Home screen has all functionality of pulling videos based off of number of likes and date uploaded. It also displays username, title, views and likes right under video. Home screen needs a lot of front end styling.
Game library API did not work as intended, so we will not on focus on that for now.

### CODE REVIEW

File: emuu-client/.gitignore [hunk 1]
```diff
@@ -24,4 +24,5 @@ yarn-error.log*
 
 serviceAccountKey.json
 .prettierignore
-src/gameTagAPI.js
\ No newline at end of file
+src/gameTagAPI.js
```
Comment (R313mon) [inline emuu-client/.gitignore:27]: Delete gam

File: emuu-client/src/components/UserProfile/Profile.js [hunk 2]
```diff
@@ -1,9 +1,43 @@
...
+    const uploadTask = uploadBytesResumable(storageRef, file);
+
+    uploadTask.on(
+      "state_changed",
+      (snapshot) => {},
+      (err) => console.log(err),
+      () => {
+        // download url
+        getDownloadURL(uploadTask.snapshot.ref).then((url) => {
+          console.log(url);
```
Comment (R313mon) [inline emuu-client/src/components/UserProfile/Profile.js]: We shouldn't be logging url, this should be stored as a variable then sent to firestore

File: emuu-client/src/components/UserProfile/Profile.js [hunk 3]
```diff
@@ -1,9 +1,43 @@
...
+    setFile(event.target.files[0]);
+  }
+  //If a user doesn't choose a file and tries to upload, error will appear
+  const handleUpload = () => {
+    if (!file) {
+      alert("Please upload an image first");
+    }
+
+    //Store into video folder in firebase storage
+    const storageRef = ref(storage, `/images/${file.name}`);
```
Comment (R313mon) [inline emuu-client/src/components/UserProfile/Profile.js]: Good use of making the storage directory as a const. This will make code easier to follow back to and read.

File: emuu-client/src/components/upload/UploadButton.js [hunk 4]
```diff
@@ -102,22 +95,20 @@ function FileUpload() {
       alert("Please upload a video first");
     }
     //Restrict file size to 40 MB ~ equivalent to 30 second video
-    if (file.size > 40 * 1024 * 1024) {
+    if (file.size > 5 * 1024 * 1024) {
```
Comment (Moamin1) [inline emuu-client/src/components/upload/UploadButton.js:104]: I believe the file size should be bigger to support larger video sizes

File: emuu-client/src/components/home/Home.js [hunk 5]
```diff
@@ -20,91 +31,102 @@ const theme = createTheme({
...
 function Home() {
+  const [topVideos, setTopVideos] = useState([]);
+  const [recentVideos, setRecentVideos] = useState([]);
+
+  async function getVideos() {
+    //Get all video data
+    const querySnapshot = await getDocs(collection(db, "Videos"));
+
+    //Create array for top videos and sort by likes
+    const querySnapshotTop = [];
```
Comment (R313mon) [inline emuu-client/src/components/home/Home.js:47]: no need to call the array "querySnapshot". Name it something easier to follow through like "TopVideos = [ ];".since a lot of the variables here are similar.

File: emuu-client/src/components/home/Home.js [hunk 6]
```diff
@@ -20,91 +31,102 @@ const theme = createTheme({
...
+      for (let j = 0; j < videos.length - 1 - i; j++) {
+        if (videos[i].data().uploadTime < videos[i + 1].data().uploadTime) {
+          let temp = videos[i];
+          videos[i] = videos[i + 1];
+          videos[i + 1] = temp;
+        }
+      }
+    }
+  }
+
```
Comment (R313mon) [inline emuu-client/src/components/home/Home.js:92]: Nice use of a bubble sort to sort videos.

**Review by Moamin1 — APPROVED** [review]
Great Job, I only found a few issues that can easily be fixed.

File: emuu-client/src/components/upload/UploadButton.js [hunk 7]
```diff
@@ -68,22 +67,16 @@ function FileUpload() {
...
-   //Gets user authentication
-   const auth = getAuth();
-   const user = auth.currentUser;
-
-
+    user_userName: userName,
+    video_title: videoTitle,
+    video_description: videoDescription,
+    video_gameTags: videoTag,
+    video_url: videoUrl,
```
Comment (R313mon) [inline emuu-client/src/components/upload/UploadButton.js]: We should switch all variables to camel case before merge to main.

File: emuu-client/src/components/upload/UploadButton.js (hunk 4 above)
Comment (R313mon) [inline emuu-client/src/components/upload/UploadButton.js:104]: This file size is too small. A 1080p 30 second video is roughly 40mb. To use 5 mb would be limited to 240p to 480p

File: emuu-client/src/components/upload/UploadButton.js [hunk 8]
```diff
@@ -133,17 +124,16 @@ function FileUpload() {
...
-            await axios
-                    .post("http://localhost:8080/auth/upload", JSON.stringify(uploadData))
-                    .then((result) => {
-                      console.log("User information is sent to firestore");
-                    });
+    //axios request to post upload information to backend
+    await axios
+      .post("http://localhost:8080/auth/upload", JSON.stringify(uploadData))
+      .then((result) => {
+        console.log("User information is sent to firestore");
```
Comment (R313mon) [inline emuu-client/src/components/upload/UploadButton.js]: Delete these console logs before we push to main.

**Review by R313mon — APPROVED** [review]
Overall, good work. Most of the functions should work logically, especially the sort videos functions. Only main issue is having variable names very similar such as querySnapshot which makes it hard to follow. Also, we should all switch our variables to camel case before we push to main.
